src/js/common/schemaform/FormIntroButtons.jsx

```diff
@@ -4,40 +4,17 @@ import PropTypes from 'prop-types';
 import ProgressButton from '../../common/components/form-elements/ProgressButton';
 
 class FormIntroButtons extends React.Component {
-  componentWillMount() {
-    // TODO: When the api is available to get a list of all the saved forms, query
-    //  for that instead and save it in the redux store
-
-    // Load the form data into the redux store if possible
-    // NOTE: This does not fill the data into the form yet; just grabs it and
-    //  saves it for later (when we want to fill in the blanks).
-    // TODO: Probably should wrap this in
-    // if (this.props.loggedIn) { ... }
-    this.props.fetchInProgressForm(this.props.form.formId, this.props.form.migrations);
-  }
-
-  // Load the form data again if we weren't logged in before but are now.
-  componentWillReceiveProps(newProps) {
-    // Ugh. So verbose.
-    let wasLoggedIn;
-    let isLoggedIn;
-    try {
-      wasLoggedIn = this.props.loggedIn;
-    } catch (e) {
-      if (e instanceof TypeError) {
-        // Probably couldn't read a property somewhere along the way
-        wasLoggedIn = false;
-      }
-    }
-    try {
-      isLoggedIn = newProps.loggedIn;
-    } catch (e) {
-      if (e instanceof TypeError) {
-        isLoggedIn = false;
-      }
-    }
-    if (!wasLoggedIn && isLoggedIn) {
-      this.props.fetchInProgressForm(this.props.form.formId, this.props.form.migrations);
+  componentWillReceiveProps = (newProps) => {
```

Comment:
I think the answer to this is moving the routing to the wrapper `SIPIntroductionPage` when we make it. Not the current one, but a HOC we can wrap our `IntroductionPage`s with that handles all the save in progress logic. Making a ticket for it now...